prow/gcloud-deployer-service-account.sh

```diff
@@ -25,13 +25,87 @@ set -o pipefail
 # a postsubmit job.
 
 # To use, point your kubeconfig at the correct cluster context and specify gcp
-# PROJECT and service account DESCRIPTION environment variables.
+# PROJECT and service account DESCRIPTION environment variables. Optionally, one can
+# supply the PROJECT_BUILD variable to attach the iam policy to the build cluster project.
 
-gcloud beta iam service-accounts create prow-deployer --project="${PROJECT}" --description="${DESCRIPTION}" --display-name="Prow Self Deployer SA"
-gcloud projects add-iam-policy-binding "${PROJECT}" --member="serviceAccount:prow-deployer@${PROJECT}.iam.gserviceaccount.com" --role roles/container.developer
-gcloud iam service-accounts keys create prow-deployer-sa-key.json --project="${PROJECT}"  --iam-account="prow-deployer@${PROJECT}.iam.gserviceaccount.com"
+# To enable prompts and run in "interactive" mode supply the "-i|--interactive" flag.
+# e.g.
+#  PROJECT="istio-testing" \
+#  PROJECT_BUILD="istio-prow-build" \
+#  DESCRIPTION="Used to deploy to the clusters in the istio-testing and istio-prow-build projects." \
+#  gcloud-deployer-service-account.sh --interactive
 
-kubectl create secret generic -n test-pods prow-deployer-service-account --from-file=service-account.json=prow-deployer-sa-key.json
+# Globals:
+PROJECT_BUILD="${PROJECT_BUILD:=''}"
+SERVICE_ACCOUNT="${SERVICE_ACCOUNT:='prow-deployer'}"
+# PROJECT => "required"
+# DESCRIPTION => "required"
 
-rm prow-deployer-sa-key.json
+# Options:
+INTERACTIVE=
 
+function cleanup() {
+  # For security reasons, delete private key regardless of exit code.
+  trap 'rm -f "$SERVICE_ACCOUNT-sa-key.json"' EXIT
+}
+
+function create_service_account() {
+  [ "$INTERACTIVE" ] && prompt "Create service-account: \"$SERVICE_ACCOUNT\" in Project: \"$PROJECT\""
```

Comment:
Why don't you move this logic into prompt?

Reply:
Updated. Thanks for the tip. That is a much cleaner interface.